### PR TITLE
coafile.rst: Add section appending docs

### PR DESCRIPTION
**Pull request by Mixih**
Document the new section appending behavior added by the secinherit
patch series.

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [ ] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [ ] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `cobot mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

File: Users/coafile.rst [hunk 1]
```diff
@@ -38,8 +38,95 @@ wide coafile which has the lowest priority. The ``default_coafile`` must
...
 this coala installation.
 
-Setting Inheritance
--------------------
+Explicit Setting Inheritance
+----------------------------
+
+Every coafile contains one or more sections. Section names are case
+insensitive. The old implicit section inheritance syntax has been deprecated
+and has been scheduled for removeal. Instead, define section inheritance
```
Comment (meetmangukiya) [inline Users/coafile.rst]: typo `removeal` :P

File: Users/coafile.rst [hunk 2]
```diff
@@ -38,8 +38,95 @@ wide coafile which has the lowest priority. The ``default_coafile`` must
...
+explicitly by naming a section in the format ``[basesection.newsection]``.
+Extra values can be appended to an inherited setting using the ``+=``
+operator.
+
+Consider the following coafile::
+
+  [all]
+  enabled = True
+  overridable = 2
+  ignore = ./vendor1
```
Comment (meetmangukiya) [inline Users/coafile.rst]: is the `./` significant or is it just to show it is current directory file

File: Users/coafile.rst [hunk 3]
```diff
@@ -38,8 +38,95 @@ wide coafile which has the lowest priority. The ``default_coafile`` must
...
+and has been scheduled for removeal. Instead, define section inheritance
+explicitly by naming a section in the format ``[basesection.newsection]``.
+Extra values can be appended to an inherited setting using the ``+=``
+operator.
+
+Consider the following coafile::
+
+  [all]
+  enabled = True
+  overridable = 2
```
Comment (meetmangukiya) [inline Users/coafile.rst]: What is overridable, just another setting name or something critical for section inheritance?

File: Users/coafile.rst [hunk 4]
```diff
@@ -38,8 +38,95 @@ wide coafile which has the lowest priority. The ``default_coafile`` must
...
+  setting1 = 1
+
+  [all.section1]
+  # setting1 is inherited
+  setting2 = 2
+
+Please only define inheritance from one section per child section::
+
+  # BAD! Will only inherit from "all", not "child" leaving "child.sub" as the
+  # new section's name
```
Comment (sils) [inline Users/coafile.rst]: I don't understand this, what do you mean here?

File: Users/coafile.rst [hunk 5]
```diff
@@ -38,8 +38,95 @@ wide coafile which has the lowest priority. The ``default_coafile`` must
...
+
+  # ok, "child" inherits from section "all" and "sub" inherits from "child"
+  # which inherits from "all"
+  [all.child]
+  setting1 += b
+
+  [child.sub]
+  setting1 = a, b, c
+
+Setting Inheritance (deprecated)
```
Comment (sils) [inline Users/coafile.rst]: let's just remve the docs for this people shouldn't use or read about it.

File: Users/coafile.rst [hunk 6]
```diff
@@ -38,8 +38,95 @@ wide coafile which has the lowest priority. The ``default_coafile`` must
...
+--------------------------------
+
+.. note::
+
+  Implicit ``default`` section inheritance has been deprecated and scheduled
+  for removal in coala version 0.11.0. This section has been kept here for
```
Comment (meetmangukiya) [inline Users/coafile.rst]: that is this release, did you mean 0.12.0?

File: Users/coafile.rst (hunk 2 above)
Comment (Mixih) [inline Users/coafile.rst]: just to show it's a dir file

File: Users/coafile.rst (hunk 3 above)
Comment (Mixih) [inline Users/coafile.rst]: nope, just to indicate it's being overridden, I just copied the last file :P

File: Users/coafile.rst (hunk 4 above)
Comment (Mixih) [inline Users/coafile.rst]: I mean you cant inherit from all and child using all.child.newsec. I agree it's confusing

File: Users/coafile.rst (hunk 5 above)
Comment (Mixih) [inline Users/coafile.rst]: what about the users of old versions?

File: Users/coafile.rst [hunk 7]
```diff
@@ -38,8 +38,95 @@ wide coafile which has the lowest priority. The ``default_coafile`` must
 lie in the coala installation directory and is valid for everyone using
 this coala installation.
 
-Setting Inheritance
--------------------
+Explicit Setting Inheritance
+----------------------------
+
+Every coafile contains one or more sections. Section names are case
```
Comment (Asnelchristian) [inline Users/coafile.rst]: `Sections'`

File: Users/coafile.rst (hunk 6 above)
Comment (Mixih) [inline Users/coafile.rst]: oh right 👍

File: Users/coafile.rst (hunk 7 above)
Comment (Mixih) [inline Users/coafile.rst]: why apostrophe?

File: Users/coafile.rst (hunk 5 above)
Comment (sils) [inline Users/coafile.rst]: we'd have to version our docs but that's not the right day to do it now, maybe directly after the release.

File: Users/coafile.rst (hunk 7 above)
Comment (Mixih) [inline Users/coafile.rst]: Seems alright to me without the apostrophe, see the old docs

File: Users/coafile.rst (hunk 5 above)
Comment (Mixih) [inline Users/coafile.rst]: alright :+1:

File: Users/coafile.rst (hunk 2 above)
Comment (sils) [inline Users/coafile.rst]: then do `vendor/` instead looks better IMO

File: Users/coafile.rst (hunk 4 above)
Comment (Asnelchristian) [inline Users/coafile.rst]: so you can't use that notation for multiple inheritance

Comment (Asnelchristian) [inline Users/coafile.rst]: if that is it, maybe you should make it clearer by saying it explicitly

**Comment by jayvdb** [comment]
ack d70f9a1

**Comment by sils** [comment]
nice good to go IMO :)

**Comment by Mixih** [comment]
@rultor merge

**Pull request merged**